### ISSUE TITLE
ASA Issues on iOS 14.2.0+ Devices 

### ISSUE THREAD

**Issue by msftradford** [Azure/azure-spatial-anchors-samples#228]
We are currently investigating an issue between the ASA client SDK and the iOS 14.2.0 release. This issue impacts all versions of the ASA SDK when deployed on devices running iOS 14.2.0. The issue manifests either as sufficiency level not incrementing while attempting to gather visual data for anchor creation or query, or as the callback for save anchor to cloud never getting raised even though the anchor was successfully saved to the cloud. This is a high priority issue for the Azure Spatial Anchors team and we will provide updates as soon as possible. Until then, please avoid upgrading your iOS devices to the 14.2 release.

**Comment by ephb** [comment]
There seems to be an additional issue we are currently investigating.
Sometimes devices running 14.2 seem to loose network connectivity after starting a watcher. The devices never receive any anchor information and no other parts of the app can connect to the network anymore.

We noticed this because we us a bug reporting that can not send anything after this happens. The iOS device logs shows ssl errors after this lockup state.

This happens in about 1 of 3 tries. It dos not matter if the watcher is started with an anchor id list or coarse relocation.

We noticed this on iOS 14.2, ASA v2.6.0, Unity 2019.4.7f1

**Comment by avi4gravity** [comment]
any resolution to this issue or is the plan to wait and check it on iOS 14.3

**Comment by ephb** [comment]
Well, one could check 14.3 right now.  I just do not want to through the hassle of upgrading and downgrading my main device.

**Comment by rodrigomas** [comment]
It is not working on 14.3. 
Same as 14.2

**Comment by msftradford** [comment]
The ASA team is still actively investigating this issue. We believe that this issue affects devices running iOS 14.2+. We will be sure to provide updates as soon as they are available. Thanks!

**Comment by cmelian** [comment]
For reference: https://docs.microsoft.com/en-us/answers/questions/162996/new-azure-spatial-anchors-270-sdk-fails-to-upload.html

We experience the same issue.
Our analysis on this reveals at least 2 different issues :

**SSL concerns of SDK backend subdomains** related to IOS14 certificate transparency policy for :
"*.events.data.microsoft.com"

For SSL certificates transparency analysis we may see here that at least 3 certificates have only a single CT Log. Apple requires now at least 2 CT for a certificate valid for less than 15 months.

https://transparencyreport.google.com/https/certificates?cert_search_auth=&cert_search_cert=p:bW9iaWxlLmV2ZW50cy5kYXRhLm1pY3Jvc29mdC5jb206dHJ1ZTpmYWxzZTo6Q0FvUUFRPT0&cert_search=include_subdomains:false;domain:mobile.events.data.microsoft.com&lu=cert_search_cert

More details on Apple CT policy :
https://support.apple.com/en-us/HT205280

When the IOS SDK makes a request, if the SSL certificates returned have less than 2 CT logs, it will be considered "not valid" by IOS.


**2nd issue identified :**
Some machines / IPs on the backend randomly won't answer to requests. Example for the IP : 52.114.75.78 which is on the pool of ips behind "mobile.events.data.microsoft.com"

![45125-capt-sslreport-mobileeventsdata](https://user-images.githubusercontent.com/2199009/101738379-5c375d80-3ac6-11eb-8ed0-a82be8ebc61d.png)

**Comment by msftradford** [comment]
We are treating this issue with the highest priority and are currently testing a set a of mitigations and fixes. If our testing goes well, a new ASA SDK 2.7.1 will be released in the first week of January to address this issue.

**Comment by ephb** [comment]
Hi @msftradford 
Any news on this?
We are badly blocked by this bug in the project we are working on since it is not even possible to buy iOS devices on 14.1 anymore and downgrading is not possible either.

**Comment by msftradford** [comment]
Hi @ephb! We have all hands on deck this week to help us test ASA SDK 2.7.1 and we are looking forward to providing an update soon. I will be sure to update this thread as soon as possible.

**Comment by msftradford** [comment]
We have just released ASA SDK 2.7.1. The release notes are located [here](https://github.com/Azure/azure-spatial-anchors-samples/releases/tag/v2.7.1). The release is intended to address this issue and unblock developers experiencing issues with iOS 14.2+.

**Comment by ephb** [comment]
Great news, I will test it today.

**Comment by rodrigomas** [comment]
Great! Thank you very much!

**Comment by git-radiusmedia** [comment]
Hello @msftradford,

I have done some tests with iOS 14.2 and the latest ASA SDK 2.7.1, and I am still facing the issues mentioned above https://github.com/Azure/azure-spatial-anchors-samples/issues/228#issuecomment-733841299. Can you please clarify whether it is fixed or any plan to fix it in the future?

Thanks in advance

**Comment by ephb** [comment]
@git-radiusmedia which region is your ASA instance running in?

**Comment by git-radiusmedia** [comment]
Europe

**Comment by msftradford** [comment]
Hello @ephb and @git-radiusmedia, thank you for bringing this to our attention. @git-radiusmedia, would you mind filing a separate issue to discuss these network connectivity issues? We'll move the discussion over to your issue. Thanks!

**Comment by msftradford** [comment]
I'm closing this out. If you are experiencing networking problems potentially related to the ASA SDK and you are deploying using Unity to devices running iOS 14.2+, please see [this issue](https://github.com/Azure/azure-spatial-anchors-samples/issues/234). Thanks!
